python/ray/experimental/slib/examples/simple.py

```diff
@@ -0,0 +1,84 @@
+import time
+import logging
+import argparse
+
+from ray.experimental.slib.streaming import *
+
+logger = logging.getLogger(__name__)
+logger.setLevel("INFO")
+
+# Test functions
+def splitter(line):
+    return line.split(' ')
+
+def filter_fn(word):
+     if "f" in word: return True
+     return False
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--input-file", required=True,
+                    help="the input text file")
+
+if __name__ == '__main__':
+
+    args = parser.parse_args()
+
+    ray.init()
+    ray.register_custom_serializer(BatchedQueue, use_pickle=True)
+    ray.register_custom_serializer(OpType, use_pickle=True)
+    ray.register_custom_serializer(PStrategy, use_pickle=True)
+
+    start = time.time()
+
+    # A Ray streaming environment with the default configuration
+    env = Environment()
+
+    # Stream represents the ouput of the filter and can be forked into other dataflows
+    stream = env.read_text_file(args.input_file) \
```

Comment:
Avoid using `\` for line continuations. Python discourages it.

The way I'd recommend, though you may find it ugly in this case, is

```python
    stream = (env.read_text_file(args.input_file)
                  .shuffle()
                  .flat_map(splitter)
                  .set_parallelism(4)
                  .filter(filter_fn)
                  .set_parallelism(2)
                  .inspect(print))
```

From https://www.python.org/dev/peps/pep-0008/

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.